### DISCUSSION TITLE
Bazel support

### DISCUSSION THREAD

**Discussion by meftunca** [stephenberry/glaze#983] [General]
Please provide support for bazel modules. Below is the relevant link
https://registry.bazel.build/
I am sharing sample basel file content for integration

```starlark
load("@rules_cc//cc:defs.bzl", "cc_library")

package(default_visibility = ["//visibility:public"])

cc_library(
    name = "glaze",
    srcs = glob(["include/glaze/**/*.hpp"]),
    hdrs = [],
    # alwayslink = True,
    copts = [

        "-DNEDBUG",
        "-std=c++20",
        "-fno-rtti",
        # set apple m2 arch
        "-march=native",
        "-mtune=native",
    ],
    includes = ["include"],
    linkopts = [],
    strip_include_prefix = "include",
)
```

**Comment by stephenberry** [comment]
What do you mean by providing bazel support, what is needed? Pardon my ignorance with bazel. If you submitted a pull request I would be happy to merge it.

**Comment by meftunca** [comment]
Hi, bazel has its own package management system and it allows developers to install bazel projects in a simpler way. Below I share with you the repo where package installation information is kept. 

central package repo: https://github.com/bazelbuild/bazel-central-registry#roles


In addition, you will need 2 files to add to your project. I will open the pull request about this until tomorrow morning.

Thank you for your interest my friend :)

**Comment by meftunca** [comment]
Hey @stephenberry , I'm sorry I couldn't make things faster for you. As the project owner, bazel has added a github app https://github.com/apps/publish-to-bcr to automate your work.  Please review it and let's complete the process together as soon as possible :)

**Reply by stephenberry** [reply]
Thanks for the info. I would much prefer a GitHub Actions approach and saw this discussed under the issues. Is it reasonable to wait until this approach is supported?

**Comment by kdowney-lot49** [comment]
@stephenberry if you are interested, I might be able to help with this one, although maybe not this weekend. ;-) I already have a working local Glaze module since it's not available in the Bazel Central Registry (BCR). I believe the right thing to do here is actually to contribute directly to BCR:

https://github.com/bazelbuild/bazel-central-registry/blob/main/docs/README.md

Then anyone can just do:

```
bazel_dep(name = "glaze", version = "7.0.2")
```

in their `MODULE.bazel`.

**Reply by stephenberry** [reply]
I'd love help with this. I haven't used Bazel, so any assistance would be appreciated.
